rrule/src/core/date_filter.rs

```diff
@@ -1,17 +1,17 @@
-use super::datetime::DateTime;
+use crate::core::utils::{collect_or_error, collect_with_error};
+use crate::core::DateTime;
 use crate::{RRuleError, WithError};
 
-pub trait DateFilter<'a, T>
+pub trait DateFilter
 where
-    &'a Self: 'a + Sized + IntoIterator<IntoIter = T>,
-    T: Iterator<Item = DateTime> + WithError,
+    Self: Sized + Iterator<Item = DateTime> + WithError,
 {
     /// Returns all the recurrences of the rrule.
     ///
     /// Limit must be set in order to prevent infinite loops.
     /// The max limit is `65535`. If you need more please use `into_iter` directly.
-    fn all(&'a self, limit: u16) -> Result<Vec<DateTime>, RRuleError> {
-        super::collect_or_error(self.into_iter(), &None, &None, true, limit)
```

Comment:
I couldn't make the old way work, so users must use `rrule.into_iter().all(...)`.
Any suggestion to revert this `into_iter` thingy change is very welcome.

Reply:
If we couldn't revert it back, we can also remove the `IntoIteratorWithCtx` trait.

Reply:
@fmeringdal 
I realized, since it's not possible to iterate over `RRule`. And `RRuleIter` will not be public.
We don't need the `DateFilter` trait at all!

I removed it locally and moved the methods to `RRuleSet`, they work and are happy!
I'll push my changes within a new commit. We can revert if you disagree.

We internally only require `all_between` for `RRuleIter` to calculate exrules. We might be able to even remove that!

Reply:
Agreed. I was also thinking that `DateFilter` could be removed if `RRule` does not implement it. It also makes the library easier to use as the user does not have to import `DateFilter` to call the methods of `RRuleSet`.